src/exchanges/gemini/GeminiExchangeAPI.ts

```diff
@@ -0,0 +1,171 @@
+                baseMaxSize: Big('1e18'),
+                quoteIncrement: Big(0.01)
+            },
+            {
+                id: 'ethbtc',
+                baseCurrency: 'ETH',
+                quoteCurrency: 'BTC',
+                baseMinSize: Big(0.001),
+                baseMaxSize: Big('1e18'),
+                quoteIncrement: Big(0.00001)
+            }
+        ];
+        return Promise.resolve(products);
+    }
+    loadMidMarketPrice(gdaxProduct: string): Promise<BigNumber.BigNumber> {
+        return this.loadTicker(gdaxProduct).then((ticker) => {
+            if (!ticker || !ticker.bid || !ticker.ask) {
+                throw new Error('Loading midmarket price failed because ticker data was incomplete or unavailable');
+            }
+            return ticker.ask.plus(ticker.bid).times(0.5);
+        });
+    }
+    loadOrderbook(gdaxProduct: string): Promise<BookBuilder> {
+        return this.loadGeminiOrderbook(gdaxProduct).then((body) => {
+            return Promise.resolve(this.buildBook(body));
+        });
+    }
+    loadGeminiOrderbook(gdaxProduct: string): Promise<string> {
+        return this.publicRequest('/book/' + gdaxProduct)
+            .then((body) => {
+                if (!(body.bids && body.asks)) {
+                    throw new Error('loadGeminiOrderbook did not return bids or asks');
+                }
+                return Promise.resolve(body);
+            });
+    }
+    loadTicker(gdaxProduct: string): Promise<Ticker> {
+        return this.publicRequest('/pubticker/' + gdaxProduct)
+            .then((ticker) => {
+                return {
+                    productId: gdaxProduct,
+                    ask: ticker.ask ? Big(ticker.ask) : undefined,
+                    bid: ticker.bid ? Big(ticker.bid) : undefined,
+                    price: Big(ticker.last || 0),
+                    time: new Date()
+                };
+            });
+    }
+    placeOrder(order: PlaceOrderMessage): Promise<LiveOrder> {
+        throw new Error('Method not implemented.');
+    }
+    cancelOrder(id: string): Promise<string> {
+        throw new Error('Method not implemented.');
+    }
+    cancelAllOrders(product: string): Promise<string[]> {
+        throw new Error('Method not implemented.');
+    }
+    loadOrder(id: string): Promise<LiveOrder> {
+        throw new Error('Method not implemented.');
+    }
+    loadAllOrders(gdaxProduct: string): Promise<LiveOrder[]> {
+        throw new Error('Method not implemented.');
+    }
+    loadBalances(): Promise<Balances> {
+        throw new Error('Method not implemented.');
+    }
+
+    private publicRequest(command: string): Promise<any> {
+        const url = `${this.apiURL}${command}`;
+        return request.get(url)
+            .accept('application/json')
+            .then((response) => {
+                if (response.status !== 200) {
+                    throw new Error('publicRequest for ' + command + ' was not successful');
```

Comment:
It may be a good idea to include the status code in the error message.

Reply:
Done.